src/Sparrow/Logging/LogginFilter.cs

```diff
@@ -0,0 +1,145 @@
+    {
+
+            private readonly Dictionary<string, List<string>> _rules = new Dictionary<string, List<string>>();
+            private readonly string[] _entryFields;
+
+            public LogginFilter()
+            {
+                _entryFields = typeof(LogEntry).GetFields().Select(p => p.Name.ToLower()).ToArray();
+                foreach (var entry in _entryFields)
+                {
+                    _rules.Add(entry, new List<string>());
+                }
+            }
+
+            public void Add(string rule)
+            {
+                string[] parts = rule.Split(':');
+                string key = parts[0];
+                string value = parts[1];
+
+                if (_entryFields.Contains(key) && value.Length > 0)
+                {
+                    _rules[key].Add(value.ToLower());
+                }
+            }
+            public void Del(string rule)
+            {
+                string[] parts = rule.Split(':');
+                string key = parts[0];
+                int index = int.Parse(parts[1]) - 1;
+
+                if (_entryFields.Contains(key) && index < _rules[key].Count)
+                {
+                    _rules[key].RemoveAt(index);
+                }
+            }
+            public string List()
+            {
+                StringBuilder sb = new StringBuilder();
+                foreach (var rule in _rules)
+                {
+                    sb.AppendLine("-" + rule.Key);
+                    for (int index = 0; index < rule.Value.Count; index++)
+                    {
+                        var r = rule.Value[index];
+                        sb.AppendLine("\t " + (index + 1) + ". " + r);
+                    }
+                }
+                return sb.ToString();
+            }
+
+            public string GetHelp()
+            {
+                StringBuilder sb = new StringBuilder();
+                sb.AppendLine("usage: filter [-add] [-del] [-list]\n");
+                sb.AppendLine("The available headers are:");
+                // Array.ForEach(entries, (entry) => sb.Append(entry));
+                foreach (var entry in _entryFields)
+                {
+                    sb.Append(entry + "  ");
+                }
+                
+                sb.AppendLine("\n-add <header>:somestring");
+                sb.AppendLine("-del <header>:rule number");
+            return sb.ToString();
+            }
+            public bool Forward(LogEntry entry)
+            {
+                bool valid = true;
+              
+                foreach (var e in _entryFields)
+                {
+                    var listOfRules = _rules[e];
+                    if (listOfRules.Count != 0)
+                    {
+                    valid = false;
+                    switch (e)
+                        {
+                        // Don't know how to do it more generic (and prettier) without the use of reflection
+                        case "at":
```

Comment:
This is bad. You are doing multiple allocations here, for the `Any` call, for the `ToLower`, etc.
This is also done elsewhere, but those happen much more rarely. Here this is called every single log entry, which means that you need to consider performance.